dali/operators/generic/slice/slice_base.cc

```diff
@@ -58,12 +58,11 @@ bool SliceBaseCpu<OutputType, InputType, Dims>::SetupImpl(std::vector<OutputDesc
   output_desc[0].type = TypeInfo::Create<OutputType>();
   output_desc[0].shape.resize(nsamples, Dims);
 
-  kmgr_.Resize<Kernel>(nthreads, nsamples);
   kernels::KernelContext ctx;
   auto in_view = view<const InputType, Dims>(input);
   for (int i = 0; i < nsamples; i++) {
     auto in_view = view<const InputType, Dims>(input[i]);
-    auto req = kmgr_.Setup<Kernel>(i, ctx, in_view, args_[i]);
+    auto req = kernel_.Setup(ctx, in_view, args_[i]);
```

Comment:
Please leave a comment about the kernel being stateless _here_. If some state is introduced, this code will be incorrect.